File: app/src/main/java/com/v2/desafionubank/di/module/ActivityModule.java
package com.v2.desafionubank.di.module;

import android.content.Context;

import com.v2.desafionubank.di.ActivityContext;
import com.v2.desafionubank.ui.BaseMain;

import dagger.Module;
import dagger.Provides;

/**
 * Created by CaioSChristino on 05/02/18.
 */

@Module
public class ActivityModule {
    private BaseMain mBaseMain;

    public ActivityModule(BaseMain baseMain) {
        mBaseMain = baseMain;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mBaseMain;
    }

    @Provides
    BaseMain provideActivity() {
        return mBaseMain;
    }
}

File: app/src/main/java/com/v2/desafionubank/di/module/ActivityModule.kt
package com.v2.desafionubank.di.module

import android.content.Context
import com.v2.desafionubank.di.ActivityContext
import com.v2.desafionubank.ui.BaseMain
import dagger.Module
import dagger.Provides



/**
 * Created by csanchez on 19/04/2018.
 */

@Module
class ActivityModule(private val mBaseMain: BaseMain) {

    @Provides
    @ActivityContext
    internal fun provideContext(): Context {
        return mBaseMain
    }

    @Provides
    internal fun provideActivity(): BaseMain {
        return mBaseMain
    }
}
